src/main/scala/io/citrine/lolo/bags/BaggedResult.scala

```diff
@@ -407,13 +408,47 @@ case class MultiPredictionBaggedResult(
   }
 }
 
+/** Enumerates the ways in which uncertainty correlation can be calculated. */
+object CorrelationMethods extends Enumeration {
+  type CorrelationMethod = Value
+
+  val Trivial, FromTraining, Bootstrap, Jackknife, JackknifeExplicit = Value
+}
+
 /**
   * Container with model-wise predictions for each label and the machinery to compute (co)variance.
   *
   * @param baggedPredictions  bagged prediction results for each label
   * @param realLabels         a boolean sequence indicating which labels are real-valued
+  * @param NibIn              the sampling matrix as (# bags) x (# training)
   */
-case class MultiTaskBaggedResult(baggedPredictions: Seq[BaggedResult[Any]], realLabels: Seq[Boolean]) extends BaggedResult[Seq[Any]] with MultiTaskModelPredictionResult {
+case class MultiTaskBaggedResult(
+                                  baggedPredictions: Seq[BaggedResult[Any]],
+                                  realLabels: Seq[Boolean],
+                                  NibIn: Vector[Vector[Int]],
+                                  trainingLabels: Seq[Seq[Any]],
+                                  trainingWeights: Seq[Double]
+                                ) extends BaggedResult[Seq[Any]] with MultiTaskModelPredictionResult {
+
+  lazy val Nib: Vector[Vector[Int]] = NibIn.transpose.map(_.map(_ - 1))
+
+  /* This matrix is used to compute the jackknife covariance */
+  lazy val NibJMat = new DenseMatrix[Double](Nib.head.size, Nib.size,
```

Comment:
I think it would be helpful to expand the docstring on what exactly these matrices are representing (e.g. are they corresponding to direct eqns in the Wager paper)?

Also, I see that they are identical to the defns in `MultiPredictionBaggedResult`. Should the logic be off-loaded to an object method and then called once to store in the `lazy val` ?

Reply:
Yeah, that all makes sense